CONTRIBUTING.md

```diff
@@ -0,0 +1,84 @@
+below).
+
+To help with triage, issues should have a descriptive title. Examples of good
+issue titles:
+
+- Require channels be unique for applications
+- "Telemetry Probe Expiry" emails sometimes don't include list of filed bugs
+
+In the issue itself, provide as much information as necessary to help someone
+reading it understand the nature of the problem (and provide feedback). For
+examples of this, look at some of the
+[fixed issues](https://github.com/mozilla/probe-scraper/issues?q=is%3Aissue+is%3Aclosed)
+filed by the project maintainers.
+
+Occasionally probe-scraper bugs are tracked inside Bugzilla, especially for issues
+which might affect other parts of the pipeline.
+
+## Opening pull requests
+
+Like issues, pull requests should have a descriptive title to help with triage.
+However there are two things that are different:
+
+- Instead of pointing out a problem, they should describe the solution
+- If a pull request fixes a specific issue, the title should specify
+  `(fixes #X)` (where X refers to the issue number)
+
+For example, a pull request to fix an issue entitled `"Telemetry Probe Expiry" emails sometimes don't include list of filed bugs` could be named `Include list of filed bugs in "Telemetry Probe Expiry" emails (fixes #1234)`.
+
+When a pull request fixes a bug in Bugzilla, prepend the bug number to the title with
+the keyword `Bug ` in the format `Bug XXXX - <one-line description>`.
+This allow the [Bugzilla PR Linker] to link to this PR automatically in bugzilla.
+For example, `Bug 1234 - Include list of filed bugs in "Telemetry Probe Expiry" emails`.
+
+As much as possible, each pull request should attempt to solve _one problem_.
+For logically separate changes, file multiple PRs.
+
+Make sure that the pull request passes continuous integration (including linter
+errors) and that there are no merge conflicts before asking for review. If you
+want some feedback on a work-in-progress (where these conditions are not yet
+met), mark your pull request as a draft.
+
+[bugzilla pr linker]: https://github.com/mozilla/github-bugzilla-pr-linker
+
+## Dangerous changes
+
+This repository is central to how ingestion and processing Telemetry data at
+Mozilla: in particular, adding new Glean repositories (`repositories.yaml` at the root
+of this repository) needs to be done with some care.
+
+Things to bear in mind:
+
+- Once probe scraper has successfully run, there is no changing or rewriting history of the metrics files, as this will cause problems downstream with [mozilla-schema-generator].
+- There is currently no provision for deleting a repository once added.
+
+As such, testing works in progress should happen locally with a probe-scraper checkout (see the "dry run" instructions in the README) and/or evaluating test pings via the [Glean Debug Ping Viewer].
+Under no circumstances should a testing application be added to this repository to "see what happens".
```

Comment:
I'm not sure if this is too strongly worded and/or restrictive. My personal feeling is that we've allowed this in the past, but probably shouldn't given the current state of our infrastructure. Could be convinced otherwise of course!

Reply:
this seems appropriate to me